hw-model/src/lib.rs

```diff
@@ -528,24 +529,22 @@ pub trait HwModel {
     /// Streams `data` to the sha512acc SoC interface. If `sha384` computes
     /// the sha384 hash of `data`, else computes sha512 hash.
     ///
-    /// `dlen` is the size of data in bytes. sha512acc does not require streamed
-    /// data to be full words.
-    ///
     /// Returns the computed digest if the sha512acc lock can be acquired.
     /// Else, returns an error.
     fn compute_sha512_acc_digest(
         &mut self,
-        data: &[u32],
-        dlen: u32,
+        data: &[u8],
         sha384: bool,
```

Comment:
nit: boolean arguments in this style should be avoided in public APIs.

Consider using either an enum or just two different methods (compute_sha512_acc_digest, compute_sha384_acc_digest).

Reply:
If you use two different methods, you can make the result type a fixed-size array.

Reply:
Switched to an enum